lib/modules/manager/sbt/extract.ts

```diff
@@ -1,368 +1,291 @@
+}
 
-  let dep: PackageDependency | null = null;
-  let scalaVersionVariable: string | null = null;
-  if (line !== '') {
-    if (isScalaVersion(line)) {
-      isMultiDeps = false;
-      const rawScalaVersion = getScalaVersion(line);
-      scalaVersion = normalizeScalaVersion(rawScalaVersion);
-      dep = {
-        datasource: MavenDatasource.id,
-        depName: 'scala',
-        packageName: 'org.scala-lang:scala-library',
-        currentValue: rawScalaVersion,
-        separateMinorPatch: true,
-      };
-    } else if (isScalaVersionVariable(line)) {
-      isMultiDeps = false;
-      scalaVersionVariable = getScalaVersionVariable(line);
-    } else if (isPackageFileVersion(line)) {
-      packageFileVersion = getPackageFileVersion(line);
-    } else if (isResolver(line)) {
-      isMultiDeps = false;
-      const url = getResolverUrl(line);
-      registryUrls.push(url);
-    } else if (isVarSeqSingleLine(line)) {
-      isMultiDeps = false;
-      const depExpr = line
-        .replace(regEx(/^.*(Seq|List|Stream)\(\s*/), '')
-        .replace(regEx(/\).*$/), '');
-      dep = parseDepExpr(depExpr, {
-        ...ctx,
-      });
-    } else if (isVarSeqMultipleLine(line)) {
-      isMultiDeps = true;
-      const depExpr = line.replace(regEx(/^.*(Seq|List|Stream)\(\s*/), '');
-      dep = parseDepExpr(depExpr, {
-        ...ctx,
-      });
-    } else if (isVarDef(line)) {
-      variables[getVarName(line)] = getVarInfo(line, ctx);
-    } else if (isVarDependency(line)) {
-      isMultiDeps = false;
-      const depExpr = line.replace(
-        regEx(/^\s*(private\s*)?(lazy\s*)?val\s[_a-zA-Z][_a-zA-Z0-9]*\s*=\s*/),
-        ''
-      );
-      dep = parseDepExpr(depExpr, {
-        ...ctx,
-      });
-    } else if (isSingleLineDep(line)) {
-      isMultiDeps = false;
-      const depExpr = line.replace(regEx(/^.*\+=\s*/), '');
-      dep = parseDepExpr(depExpr, {
-        ...ctx,
-      });
-    } else if (isPluginDep(line)) {
-      isMultiDeps = false;
-      const rightPart = line.replace(
-        regEx(/^\s*(addSbtPlugin|addCompilerPlugin)\s*\(/),
-        ''
-      );
-      const depExpr = rightPart.replace(regEx(/\)\s*$/), '');
-      dep = parseDepExpr(depExpr, {
-        ...ctx,
-        depType: 'plugin',
-      });
-    } else if (isDepsBegin(line)) {
-      isMultiDeps = true;
-    } else if (isMultiDeps) {
-      const rightPart = line.replace(regEx(/^[\s,]*/), '');
-      const depExpr = rightPart.replace(regEx(/[\s,]*$/), '');
-      dep = parseDepExpr(depExpr, {
-        ...ctx,
-      });
+const sbtPackageMatch = q
+  .opt<Ctx>(q.opt(q.sym<Ctx>('lazy')).sym('val').sym().op('='))
+  .alt(simpleDependencyMatch, versionedDependencyMatch)
+  .opt(
+    q.alt<Ctx>(
+      q.sym<Ctx>('classifier').str(depTypeHandler),
+      q.op<Ctx>('%').sym(depTypeHandler),
+      q.op<Ctx>('%').str(depTypeHandler)
+    )
+  )
+  .handler(depHandler);
+
+const sbtPluginMatch = q
+  .sym<Ctx>('addSbtPlugin')
```

Comment:
There's `addCompilerPlugin` should be able to match the plugin too.

https://github.com/renovatebot/renovate/pull/17006/files